cray_product_catalog/catalog_update.py

```diff
@@ -172,10 +187,16 @@ def update_config_map(data: dict, name, namespace):
             LOGGER.exception("Error calling read_namespaced_config_map")
 
             # ConfigMap doesn't exist yet
-            if err.status == ERR_NOT_FOUND:
+            if err.status != ERR_NOT_FOUND:
+                raise   # unrecoverable
+            if name == CONFIG_MAP:
+                # If main ConfigMap is not found wait until it is available
                 LOGGER.warning("ConfigMap %s/%s doesn't exist, attempting again", namespace, name)
-                continue
-            raise  # unrecoverable
+            else:
+                # If product ConfigMap is not available then create
+                LOGGER.warning("Product ConfigMap %s/%s doesn't exist, attempting to create", namespace, name)
+                create_config_map(api_instance, name, namespace)
```

Comment:
If we fail to create the ConfigMap in `create_config_map`, does it really make sense to keep trying to create the ConfigMap in this loop? It seems like this would most likely be due to some reason which would not be resolved upon a retry, especially given that the `api_instance` already has retries configured.

Reply:
We discussed this on a call this morning/evening. We believe it is sufficient to just fail and break out of this loop if there is a failure to create the product config map. There are already retries on the K8s client. We don't think it's likely to be a recoverable error if this fails.